src/reporter/dwcReporter.js

```diff
@@ -0,0 +1,354 @@
+  };
+  
+  if (this.collector && this.collector.overview && this.collector.overview.meta && this.collector.overview.meta.runtime) {
+    this.instance[sessionId].baseInformation.desiredCapabilities = {};
+
+    this.instance[sessionId].baseInformation.desiredCapabilities.browserName = this.collector.overview.meta.runtime.browserName;
+    this.instance[sessionId].baseInformation.desiredCapabilities.browserVersion = this.collector.overview.meta.runtime.browserVersion;
+    this.instance[sessionId].baseInformation.desiredCapabilities.platform = this.collector.overview.meta.runtime.platformName;
+    this.instance[sessionId].baseInformation.desiredCapabilities.screenResolution = this.collector.overview.meta.runtime.platformResolution;
+  }
+
+  if (process.env.STAGE) {
+    this.instance[sessionId].baseInformation.executedOnStage = process.env.STAGE;
+  }
+
+  if (process.env.GITHUB_REPO_NAME) {
+    this.instance[sessionId].baseInformation.githubRepo = process.env.GITHUB_REPO_NAME;
+  }
+
+  if (process.env.GITHUB_REPO_URL) {
+    this.instance[sessionId].baseInformation.githubRepoUrl = process.env.GITHUB_REPO_URL;
+  }
+
+  if (process.env.BUILD_URL) {
+    this.instance[sessionId].baseInformation.buildLink = process.env.BUILD_URL;
+  }
+
+  this.instance[sessionId].reportTestRun = this.instance[sessionId].baseInformation;
+  this.instance[sessionId].reportTestRun.status = status.running;
+  
+  await this.sync(this.instance[sessionId]);
+};
+
+DwcReporter.prototype._asyncSuiteFinished = async function(instance){
+  if (instance.stateCounts.failed > 0) instance.passed = false;
+  const report = instance.baseInformation;
+
+  if (instance.passed) {
+    report.status = status.success;
+  } else {
+    report.status = status.failed;
+  }
+
+  instance.reportTestRun = report;
+  await this.sync(instance);
+};
+
+DwcReporter.prototype.jasmineStarted = function() {
+  var that = this;
+  this.instance = {};
+  this.suiteInfo = {};
+  this.suiteInfOneTime = {};
+  this.suitePrevSession={};
+
+  beforeEach(async function() { 
+    var session = await browser.driver.getSession();
+    const sessionId = session.getId();
+    if (that.suitePrevSession && that.suitePrevSession.sessionId) {
+      await that._asyncSuiteFinished(that.suitePrevSession.instance);
+      that.suitePrevSession.sessionId = null;
+    }
+    if (!that.suiteInfOneTime[sessionId]) {
+      await that._asyncSuiteStarted(that.suiteInfo, sessionId, browser);
+      that.suiteInfOneTime[sessionId] = true;
+      that.suitePrevSession.sessionId = null; 
+    }
+  });
+
+  afterAll(async function() {
+    var session = await browser.driver.getSession();
+    const sessionId = session.getId();
+    await that._asyncSuiteFinished(that.instance[sessionId]);
+    that.suiteInfOneTime[sessionId] = true;       
+  });
+
+  jasmine.getEnv().addReporter(new function () {
+    this.suiteStarted = async function(result){
+      var session = await browser.driver.getSession();
```

Comment:
I don't think we need the session handling in this reporter, this is necessary for multi-thread operation but we will handle this in the statistics collector